lib/parse-template.js

```diff
@@ -2,9 +2,13 @@
 const Transform = require('./transform');
 
 module.exports = function parseTemplate (handledTags, insertBeforePipeTags) {
-    return (baseTemplate, childTemplate) => new Promise((resolve) => {
-        const transform = new Transform(handledTags, insertBeforePipeTags);
-        const serializedList = transform.applyTransforms(baseTemplate, childTemplate);
-        resolve(serializedList);
+    return (baseTemplate, childTemplate) => new Promise((resolve, reject) => {
+        try {
+            const transform = new Transform(handledTags, insertBeforePipeTags);
+            const serializedList = transform.applyTransforms(baseTemplate, childTemplate);
+            resolve(serializedList);
+        } catch (e) {
+            reject(e);
+        }
```

Comment:
as you said Transforms should not throw any error and Its handled in the lib itself.. Just to be extra sure to catch errors happening while parsing `templates`, Which shouldn't happen in any case.